Introduce SwapRequestHandler for Bob

thomaseizinger opened the pull request:
Resolves #417.

This is the 2nd preparation step for doing #380. After this, we only need to "fill-in" the instantiation of the state machine in the `SwapRequestHandler` for Alice and Bob.

thomaseizinger committed:
Introduce Bob's SwapRequestHandler

thomaseizinger committed:
Bob stores metadata for received swap requests

thomaseizinger committed:
Properly implement ERC20 assets for BAM

thomaseizinger committed:
Add placeholder method for starting state machine

thomaseizinger committed:
Reduce duplication when sending swap request

thomaseizinger committed:
Rename SwapRequestsHandler to SwapRequestHandler

LLFourn reviewed (approved):
Nice work. Approving but would prefer the enums be renamed to ..`SwapRequestKind` rather than `SwapRequests` (and wherever else you feel it would be appropriate on Asset and Ledger etc).

thomaseizinger committed:
Use *Kind for enums that represent variants

Instead of the very similar looking *s convention, we use *Kind to
name enums that represent possible combinations of a struct.

D4nte reviewed (commented):
First pass

D4nte reviewed (changes requested):
Couple of comment regarding `unwrap()`.

thomaseizinger committed:
Remove unnecessary newlines
